libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/BaseInvokeDialog.cs

```diff
@@ -95,8 +96,17 @@ protected virtual Dialog ResolveDialog(DialogContext dc)
 
             // NOTE: we want the result of the expression as a string so we can look up the string using external FindDialog().
             var se = new StringExpression($"={this.Dialog.ExpressionText}");
-            var dialogId = se.GetValue(dc.State);
-            return dc.FindDialog(dialogId ?? throw new InvalidOperationException($"{this.Dialog.ToString()} not found."));
+            var dialogId = se.GetValue(dc.State) ?? throw new InvalidOperationException($"{this.Dialog.ToString()} not found.");
+            var dialog = dc.FindDialog(dialogId);
+            var resourceExplorer = dc.Context.TurnState.Get<ResourceExplorer>();
+            
+            if (resourceExplorer != null && dialog == null)
```

Comment:
nit: 
since in most cases the dialog will be found, and in rare cases the dialog will not be found, we could retrieve ResourceExplorer only when dialog == null